pkg/controller/quota/quota.go

```diff
@@ -124,26 +124,29 @@ func EnsureQuotaRequest(req router.Request, resp router.Response) error {
 // addContainers adds the number of containers and accounts for the scale of each container.
 func addContainers(containers map[string]v1.Container, quotaRequest *adminv1.QuotaRequestInstance) {
 	for _, container := range containers {
-		quotaRequest.Spec.Resources.Containers += replicas(container.Scale)
+		quotaRequest.Spec.Resources.Containers += int(replicas(container.Scale))
 	}
 }
 
 // addCompute adds the compute resources of the containers passed to the quota request.
 func addCompute(containers map[string]v1.Container, appInstance *v1.AppInstance, quotaRequest *adminv1.QuotaRequestInstance) {
 	// For each workload, add their memory/cpu requests to the quota request
 	for name, container := range containers {
-		var requirements corev1.ResourceRequirements
-		if specific, ok := appInstance.Status.Scheduling[name]; ok {
-			requirements = specific.Requirements
-		} else if all, ok := appInstance.Status.Scheduling[""]; ok {
-			requirements = all.Requirements
+		var cpu, memory resource.Quantity
+		if specific, ok := appInstance.Status.ResolvedOfferings.Containers[name]; ok {
+			memory = *resource.NewQuantity(z.Dereference(specific.Memory), resource.BinarySI)
+			cpu = *resource.NewMilliQuantity(z.Dereference(specific.CPU), resource.DecimalSI)
+		} else if all, ok := appInstance.Status.ResolvedOfferings.Containers[""]; ok {
+			cpu = *resource.NewMilliQuantity(z.Dereference(all.CPU), resource.DecimalSI)
+			memory = *resource.NewQuantity(z.Dereference(all.Memory), resource.BinarySI)
 		}
 
-		// Add the memory/cpu requests to the quota request for each container at the scale specified
-		for i := 0; i < replicas(container.Scale); i++ {
-			quotaRequest.Spec.Resources.CPU.Add(requirements.Requests["cpu"])
-			quotaRequest.Spec.Resources.Memory.Add(requirements.Requests["memory"])
-		}
+		// Multiply the memory/cpu requests by the scale of the container
+		cpu.Mul(replicas(container.Scale))
```

Comment:
This was previously in the ComputeClass quota PR but I just brought it here instead.